### PR TITLE
Return a friendly error if github API responds with status code 401

### LINKED ISSUE #1327: Bad cargo error message for adding owners if GitHub token is invalid
Moved here from rust-lang/cargo#5297.

What happened:

Niko tried to add a team as a github owner for the lalrpop crate:

```bash
>  cargo owner --add github:lalrpop:publish
    Updating registry `https://github.com/rust-lang/crates.io-index`
error: failed to invite owners to crate lalrpop: failed to get a 200 OK response, got 502
headers:
    HTTP/1.1 100 Continue



    HTTP/1.1 502 Bad Gateway

    Connection: keep-alive

    Server: nginx

    Date: Thu, 05 Apr 2018 08:47:57 GMT

    Content-Type: text/html

    Content-Length: 166

    Strict-Transport-Security: max-age=31536000

    Via: 1.1 vegur



body:
<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

In the papertrail logs, I was able to see:

```
Apr 05 04:37:33 crates-io app/web.2: ERROR 2018-04-05T08:37:33Z: conduit_log_requests: 127.0.0.1:51462 [2018-04-05T08:37:33Z] Put /api/v1/crates/lalrpop/owners - 70ms 500: didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} 
Apr 05 04:37:33 crates-io app/web.2: 2018/04/05 08:37:33 [error] 100#0: *132269 upstream prematurely closed connection while reading response header from upstream, client: 10.69.109.160, server: _, request: "PUT //api/v1/crates/lalrpop/owners HTTP/1.1", upstream: "http://127.0.0.1:8888//api/v1/crates/lalrpop/owners", host: "crates.io" 
Apr 05 04:37:33 crates-io heroku/router: at=info method=PUT path="//api/v1/crates/lalrpop/owners" host=crates.io request_id=e9608c5b-92b9-4e6b-ac32-ce5f5c0ba811 fwd="209.6.50.191" dyno=web.2 connect=1ms service=90ms status=502 bytes=356 protocol=https 
Apr 05 04:47:57 crates-io heroku/router: at=info method=PUT path="//api/v1/crates/lalrpop/owners" host=crates.io request_id=ea1e4af3-3993-43fb-be9b-a4feb7e2fff0 fwd="209.6.50.191" dyno=web.2 connect=0ms service=73ms status=502 bytes=356 protocol=https 
Apr 05 04:47:57 crates-io app/web.2: ERROR 2018-04-05T08:47:57Z: conduit_log_requests: 127.0.0.1:60820 [2018-04-05T08:47:57Z] Put /api/v1/crates/lalrpop/owners - 52ms 500: didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} 
Apr 05 04:47:57 crates-io app/web.2: 2018/04/05 08:47:57 [error] 100#0: *138593 upstream prematurely closed connection while reading response header from upstream, client: 10.151.201.12, server: _, request: "PUT //api/v1/crates/lalrpop/owners HTTP/1.1", upstream: "http://127.0.0.1:8888//api/v1/crates/lalrpop/owners", host: "crates.io" 
```

tl;dr github is saying "bad credentials".

Niko hadn't signed in to crates.io in a long time (ever?) so his github token was invalid. We should detect this error and direct the user to sign in to crates.io in the browser to get a new token in this case.

### PR DESCRIPTION
**Pull request by jtgeibel**
We already have custom logic for handling 403 responses from github.  This handles a 401 status in the same way.

Fixes #1327

### CODE REVIEW

**Comment by jtgeibel** [comment]
bors: r+
